controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/LoadBalancerMaintainer.java

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.maintenance;
+
+import com.google.common.base.Strings;
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.ClusterSpec;
+import com.yahoo.config.provision.HostName;
+import com.yahoo.log.LogLevel;
+import com.yahoo.vespa.hosted.controller.Application;
+import com.yahoo.vespa.hosted.controller.Controller;
+import com.yahoo.vespa.hosted.controller.LockedApplication;
+import com.yahoo.vespa.hosted.controller.api.identifiers.DeploymentId;
+import com.yahoo.vespa.hosted.controller.api.integration.configserver.LoadBalancer;
+import com.yahoo.vespa.hosted.controller.api.integration.dns.NameService;
+import com.yahoo.vespa.hosted.controller.api.integration.dns.RecordData;
+import com.yahoo.vespa.hosted.controller.api.integration.dns.RecordName;
+import com.yahoo.vespa.hosted.controller.api.integration.zone.ZoneId;
+import com.yahoo.vespa.hosted.controller.application.Deployment;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class LoadBalancerMaintainer extends Maintainer {
+
+    private static final Logger log = Logger.getLogger(LoadBalancerMaintainer.class.getName());
+    private static final String IGNORE_ENDPOINT_VALUE = "default";
+
+    private final NameService nameService;
+
+    public LoadBalancerMaintainer(Controller controller,
+                                  Duration interval,
+                                  JobControl jobControl,
+                                  NameService nameService) {
+        super(controller, interval, jobControl);
+        this.nameService = nameService;
+    }
+
+    @Override
+    protected void maintain() {
+        // update application object with load balancer information
+        controller().applications().asList().forEach(this::updateApplicationLoadBalancers);
+
+        // Create or update cnames
+        List<Application> applications = controller().applications().asList();
+        applications.forEach(this::registerLoadBalancerEndpoint);
+
+        // Delete removed application rotations
+        // TODO
+    }
+
+    private void updateApplicationLoadBalancers(Application application) {
+        // Get a list of all load balancers for this applications (for all zones and clusters)
+        Map<ZoneId, List<LoadBalancer>> applicationZoneLoadBalancers = new HashMap<>();
+        for (ZoneId zoneId : application.deployments().keySet()) {
+            try {
+                applicationZoneLoadBalancers.put(zoneId, controller().applications().configServer().getLoadBalancers(new DeploymentId(application.id(), zoneId)));
+            } catch (Exception e) {
+                log.log(LogLevel.INFO,
```

Comment:
Warning? Also consider adding `"Retrying in " + maintenanceInterval()` to the message.